addons/mrp/models/mrp_production.py

```diff
@@ -1208,10 +1209,12 @@ def _get_origin(self):
             origin = '%s,%s' % (origin, self.name)
         return origin
 
-    def _set_qty_producing(self, pick_manual_consumption_moves=True):
+    def _set_qty_producing(self, pick_manual_consumption_moves=True, user_changed=False):
         if self.product_id.tracking == 'serial':
             qty_producing_uom = self.product_uom_id._compute_quantity(self.qty_producing, self.product_id.uom_id, rounding_method='HALF-UP')
-            if qty_producing_uom != 1:
+            if qty_producing_uom == 0 and user_changed:
+                self.qty_producing = self.product_id.uom_id._compute_quantity(0, self.product_uom_id, rounding_method='HALF-UP')
+            elif qty_producing_uom != 1:
```

Comment:
This change doesn't work when doing the same flow in the barcode app for MOs. Also it's a bit hacky, maybe you can do something within  like:

```
if qty_producing_uom != 1 and not (qty_producing_uom == 0 and self._origin.qty_producing != self.qty_producing):
    self.qty_producing = self.product_id.uom_id._compute_quantity(1, self.product_uom_id, rounding_method='HALF-UP')
```

The self._origin is the previous value and it will indicate if the qty was manually changed to 0

This is also a bit hacky, but it is slightly cleaner